org.eclipse.scanning.device.ui/src/org/eclipse/scanning/device/ui/composites/MultiStepComposite.java

```diff
@@ -102,17 +102,33 @@ public boolean updateUnits(Object value) {
 	public void setScannableService(IScannableDeviceService service, MultiStepModel model) throws ScanningException {
```

Comment:
I haven looked how this code works but it looks like this method is called via OSGi? Its generally not a good idea to do work in these methods?